File: iter_demo.py
class Repeat:
    def __init__(self, ranges):
        pass

    def __iter__(self):
        return self

    def __next__(self):
        raise StopIteration

class Demo:
    def __init__(self):
        self._max = 10

    def __iter__(self):
        # print("iter init")
        self._idx = 0
        return self

    def __next__(self):
        # print("iter next")
        self._idx += 1

        if self._idx > self._max:
            raise StopIteration

        return self._idx

def test():
    ranges = [
        ("i", 2, 3),
        ("j", 1, 2),
    ]

    # repeator = Repeat(ranges)
    repeator = Demo()

    for info in repeator:
        print(info)

if __name__ == "__main__":
    test()

File: iter_demo.c
#include<stdio.h>
void demo(void);
int main(void)
{
	char[3][3]={{i},{2},{3},\
		    {j},{2},{3},\
		    {k},{3},{5}}
	demo();
}

void Repeat(char[] *ranges)
{
	
}





void demo(void)
{
	int i, j, k;
	for(i=2; 2<=i&&i<=3; i++){
		for(j=2; 2<=j&&j<=3; j++){
			for(k=3; 3<=k&&k<=5; k++){
				printf("i %d j %d k %d\n",i,j,k);
			}
		}
	}
}
